evm/src/trace/identifier/etherscan.rs

```diff
@@ -57,13 +106,17 @@ impl TraceIdentifier for EtherscanIdentifier {
             for (addr, _) in addresses {
                 fetcher.push(*addr);
             }
-
             let fut = fetcher
-                .map(|(address, label, abi)| AddressIdentity {
-                    address,
-                    label: Some(label.clone()),
-                    contract: Some(label),
-                    abi: Some(Cow::Owned(abi)),
+                .map(|(address, label, abi, source_code, optimization, runs, version)| {
+                    self.contracts
+                        .insert(label.clone(), (source_code, optimization, runs, version));
```

Comment:
I think we should do this keyed by address instead - if we have multiple addresses with the same label (which IIRC is just the contract name for this identifier) then we might overwrite info for two contracts with the same name, but with different implementations. E.g. if you interact with two tokens and both of them show up as "Token", then only one of them would be decoded... I think